src/main/java/org/opentestsystem/ap/ims/service/ContentSpecIdUtil.java

```diff
@@ -0,0 +1,153 @@
+package org.opentestsystem.ap.ims.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.opentestsystem.shared.contentspecid.ContentSpecId;
+import org.opentestsystem.shared.contentspecid.ContentSpecIdConverter;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecFormat;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecGrade;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecSubject;
+import org.opentestsystem.shared.contentspecid.exceptions.ErrorSeverity;
+import org.opentestsystem.shared.contentspecid.exceptions.ValidationException;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.opentestsystem.shared.contentspecid.ContentSpecIdBuilder.getBuilder;
+import static org.opentestsystem.shared.contentspecid.enums.ContentSpecSubject.MATH;
+
+/**
+ * Builds a content spec ID from component parts and formats it into an Enhanced syntax string.
+ */
+@Component
+@Slf4j
+public class ContentSpecIdUtil {
+    private final ContentSpecIdConverter converter;
+
+    public ContentSpecIdUtil(ContentSpecIdConverter converter) {
+        this.converter = converter;
+    }
+
+    public ContentSpecIdUtil() {
```

Comment:
Why is this constructor necessary?  If this is a Spring component then the converter should be something that is in the Spring container.  If it isn't maybe we should create a Configuration class in the library or here to wire this class?

Reply:
Hmm, interesting. I have the extra constructor because I wanted to be able to use a mock in tests. The content spec ID library classes are not Spring components, but I'll look into a configuration class as you suggest.

Reply:
So should this even be a @Component or should the class that needs this instantiate it?  Or you could probably make all the methods in this class static since it doesn't deal with state.